### PR TITLE
Fix tight_layout for WCSAxes

### PR DESCRIPTION
**Pull request by dstansby**
Without doing a major refactoring of WCSAxes to make things work more like they do in Matplotlib, this fixes https://github.com/astropy/astropy/issues/8703

### CODE REVIEW

File: astropy/visualization/wcsaxes/core.py [hunk 1]
```diff
@@ -617,6 +617,8 @@ def get_tightbbox(self, renderer, *args, **kwargs):
         if not self.get_visible():
             return
 
+        # Do a draw to populate the self._bboxes list
+        self.draw_wcsaxes(renderer)
```
Comment (Cadair) [inline astropy/visualization/wcsaxes/core.py:621]: How was this line not hit by a figure test? Are we not getting coverage out of circle?

**Comment by Cadair** [comment]
Should we write a figure test for this?

**Comment by dstansby** [comment]
Given this didn't change any existing tests, sounds good - I'm not sure which of the files under `wcsaxes/tests` it's best to add the tests to though, any thoughts?

**Comment by Cadair** [comment]
pick one :see_no_evil:, the one with the most useful fixtures is probably the best, maybe `test_images.py`?

**Comment by dstansby** [comment]
I think this is worth pursusing - I don't have time to figure out how to update fiugre tests in astropy at the moment, but if someone else has the motivation to feel free to push to my branch or create a new branch+PR off this branch.

**Comment by astrofrog** [comment]
I will need to rebase once https://github.com/astropy/astropy/pull/12630 is merged to get the latest infrastructure changes

**Review by astrofrog — APPROVED** [review]

**Comment by pllim** [comment]
Hmm... Did you break the autobackport bot? lol

@meeseeksdev backport to v5.0.x

@meeseeksdev backport to v5.1.x

**Comment by dstansby** [comment]
Thanks a lot @astrofrog for finishing off my PRs!

**Pull request merged**